Userland/Libraries/LibWeb/FileAPI/Blob.cpp

```diff
@@ -96,6 +97,11 @@ ErrorOr<ByteBuffer> process_blob_parts(Vector<BlobPart> const& blob_parts, Optio
                 // 2. If the endings member of options is "native", set s to the result of converting line endings to native of element.
                 if (options.has_value() && options->endings == Bindings::EndingType::Native)
                     s = TRY(convert_line_endings_to_native(s));
+                
+                // s is a USVString, which means unpaired surrogates have to be replaced.
+                // MDN says this should be the same algorithm as String.prototype.toWellFormed().
+                // https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob#blobparts
+                s = JS::to_well_formed_string(JS::Utf16String::create(s));
```

Comment:
If I understand correctly, this should be done by the IDL Generator as the String variant of `BlobPart` is a [USVString](https://webidl.spec.whatwg.org/#idl-USVString) which should use: `convert_to_scalar_value_string` from `Userland/Libraries/LibWeb/Infra/Strings.h` in the generated code `BlobConstructor.cpp` and `FileConstructor.cpp`